app/controllers/admin/vouchers_controller.rb

```diff
@@ -26,7 +26,10 @@ def create
     private
 
     def load_enterprise
-      @enterprise = Enterprise.find_by(permalink: params[:enterprise_id])
+      @enterprise = OpenFoodNetwork::Permissions
+        .new(spree_current_user)
+        .editable_enterprises
+        .find_by(permalink: params[:enterprise_id])
```

Comment:
Side comment..
It looks funny to see permalink coming from  `params[:enterprise_id]`. I can see this is common across the whole admin interface.. but I have to admit I can't see why!